### PR TITLE
PLATFORM-1945_Analytics

### PR DESCRIPTION
**Pull request by erpankajpatel**
- added analytics for flutter

### CODE REVIEW

File: example/lib/constants/configs.dart
Comment (chiragphyllo) [inline example/lib/constants/configs.dart]: @erpankajpatel , we can remove the secret & id from here

File: example/lib/constants/configs.dart
Comment (chiragphyllo) [inline example/lib/constants/configs.dart]: We can switch back to sandbox

**Review by chiragphyllo — CHANGES_REQUESTED** [review]

File: android/build.gradle
Comment (chiragphyllo) [inline android/build.gradle]: What is this URL for ?

File: example/lib/constants/configs.dart
Comment (erpankajpatel) [reply example/lib/constants/configs.dart]: @chiragphyllo updated

File: example/lib/constants/configs.dart
Comment (erpankajpatel) [reply example/lib/constants/configs.dart]: @chiragphyllo updated

File: android/build.gradle
Comment (erpankajpatel) [reply android/build.gradle]: @chiragphyllo its for snapshot version , for android first we are testing with Snapshot version in android .

File: lib/src/verison_constants.dart
Comment (akshaybhalotia) [inline lib/src/verison_constants.dart:6]: Can we take these parameters from the iOS SDK and Android SDK directly instead of hardcoding?

**Review by akshaybhalotia — CHANGES_REQUESTED** [review]

File: lib/src/verison_constants.dart
Comment (erpankajpatel) [reply lib/src/verison_constants.dart:6]: @akshaybhalotia yes we can do with bridge method between , as per PRD we need to define flutter and other version as well .
for example if app running in android we can get only android sdk version not ios sdk , because we can bridge method which depend on OS running (ios/Android) . due to this complexity i have added like hardcoded . same in react native as well .

Comment (akshaybhalotia) [reply lib/src/verison_constants.dart:6]: Got it, looks fine. Please make sure that we add this step to our release process that this file needs to be updated with each release.

Comment (erpankajpatel) [reply lib/src/verison_constants.dart:6]: yes sure

File: android/build.gradle
Comment (chiragphyllo) [inline android/build.gradle]: @erpankajpatel , once the android goes, live we will update it to original version rather than SNAPSHOT?

File: example/lib/phyllo_provider.dart
Comment (chiragphyllo) [inline example/lib/phyllo_provider.dart:96]: Can you print the sample output here, just for review?

**Review by chiragphyllo — APPROVED** [review]

File: CHANGELOG.md
Comment (akshaybhalotia) [inline CHANGELOG.md]: Put this changelog

- Added `version` method to help with debugging.
- More robust internal logging and debugging mechanisms.

**Review by akshaybhalotia — CHANGES_REQUESTED** [review]

File: android/build.gradle
Comment (erpankajpatel) [reply android/build.gradle]: @chiragphyllo once we will merge from release_v0.3.2 to main it will be change after release Android SDK .

File: example/lib/phyllo_provider.dart
Comment (erpankajpatel) [reply example/lib/phyllo_provider.dart:96]: [log] version: {connect_flutter_sdk_version: 0.3.2, connect_android_sdk_version: 0.3.1, min_supported_android_version: 21, max_supported_android_version: 31, connect_ios_sdk_version: 0.3.1, min_supported_ios_version: 12.0, max_supported_ios_version: 16.0*}

File: CHANGELOG.md
Comment (erpankajpatel) [reply CHANGELOG.md]: @akshaybhalotia ok sure , let me change it

Comment (erpankajpatel) [reply CHANGELOG.md]: @akshaybhalotia updated this changes

File: android/build.gradle
Comment (akshaybhalotia) [reply android/build.gradle]: Is Android SDK not released yet with analytics changes? If not, we should hold this release also since it will be hard to get users to update their SDKs again and again because of our internal changes.

**Review by akshaybhalotia — APPROVED** [review]

**Pull request merged**
